### ISSUE TITLE
Linux: Building with current instructions causes improper behavior when performing actions related to loading/saving files in the Editor

### ISSUE THREAD

**Issue by LB-ArturSiewniak** [o3de/o3de#5956]
**Describe the bug**
If the engine is built using official source config/build instructions (https://o3de.org/docs/welcome-guide/setup/setup-from-github/building-linux/) it affects actions related to saving or loading assets. The following issue will showcase it by listing two different config/build/project and compare different actions/issues as they occur on both of them.

I will call these builds A and B respectively, where _**A will be the official instruction set**_, and _**B will be the older instruction set**_.
_**A (official - https://o3de.org/docs/welcome-guide/setup/setup-from-github/building-linux/ - Source engine commands):**_
```
mkdir $HOME/o3de-packages
python/get_python.sh
cmake -B build/linux -S . -G "Ninja Multi-Config" -DLY_3RDPARTY_PATH=$HOME/o3de-packages
cmake --build build/linux --target Editor --config profile -j <number of parallel build tasks> (on my system cmake --build build/linux --target Editor --config profile -j 6)
scripts/o3de.sh register --this-engine

After that the Project Manager is launched using the o3de executable under /home/o3de/build/linux/bin/profile/. A new project configured with all gems enabled is created and successfully built via the Project Manager and launched from it.
```
_**B (old instruction set):**_
```
~/o3de/python/get_python.sh
~/o3de/scripts/o3de.sh register --this-engine
cd ~/o3de/AutomatedTesting
cmake -B build/linux -S . -G "Ninja Multi-Config" -DCMAKE_C_COMPILER=clang-12 -DCMAKE_CXX_COMPILER=clang++-12
cmake --build build/linux --config profile --target AutomatedTesting.GameLauncher Editor -j 6

After that the Project Manager is launched using the o3de executable under /home/o3de/AutomatedTesting/build/linux/bin/profile/. AutomatedTesting is then successfully rebuilt in the Project Manager and launched from it.
```

**Comparison of different actions**:
1. Issue https://github.com/o3de/o3de/issues/5950:
* On build A the Editor freezes when the script with a Math node with a number assigned in one of the Number fields is used in an entity with a Script Canvas component. Please refer to the Issue5950OnOtherBuild.mp4 video in the **Videos** section.
* On build B the issue occurs as described in the https://github.com/o3de/o3de/issues/5950.
2. Issue https://github.com/o3de/o3de/issues/5928:
* On build A the softlock occurs on every script save attempt.
* On build B the softlock occurs only once and after restarting the Editor it does not occur anymore.
3. Creating an .inputbindings file:
* On build A if the user creates an .inputbindings file, it is not visible in the Editor until it is restarted. Please refer to the CreatingAnInputBindingsFIle.mp4 video in the **Videos** section.
* On build B creating an .inputbindings file works without issue.
4. Generating a physxmesh on an .fbx asset:
* On build A if the user attempts to generate a physxmesh for an .fbx asset, the Editor freezes. It should be noted that the physxmesh is properly generated after restarting the Editor. Please refer to the CreatingPhysxmesh.mp4 video in the **Videos** section.
* On build B the Editor either freezes only on the first physxmesh generation and works without issue afterwards, or does not freeze at all.

**Steps to reproduce**
1. Follow the build A instruction and reproduce actions described listed under the **Comparison of different actions** section.
2. Follow the build B instruction and reproduce actions described listed under the **Comparison of different actions** section.

**Expected behavior**
Following the official instruction set for configuration and building does not cause any additional issues compared to a different method.

**Actual behavior**
Following the official instruction set for configuration and building seems to cause additional issues or exaggerate existing ones.

**Videos**

https://user-images.githubusercontent.com/86952082/143594676-c02898fd-a2c6-4054-a933-a5602d61636b.mp4

https://user-images.githubusercontent.com/86952082/143594715-5f6be4a5-e7f2-43ed-b2c5-acefae91903a.mp4

https://user-images.githubusercontent.com/86952082/143594819-ddbde913-0967-400e-a240-3a0a39c573ba.mp4

**Found in Branch**
Stabilization_2110 (99d0c39)

**Desktop**
Device: PC
OS: Linux
Version: Ubuntu 20.04.3 LTS
CPU AMD Ryzen 5 3600
GPU NVIDIA GeForce RTX 2060 SUPER
Memory 16GB

**Comment by LB-ArturSiewniak** [comment]
Thanks in advance for taking the time to read through this issue. I would like to ask as well - should the issues showcased be logged as separate issues if deemed to not be caused by the used building instructions?

**Comment by LB-MateuszSzweda** [comment]
It is worth mentioning that the .inputbings/physxmesh/.scriptcanvas assets that are initially failing in Asset Processor upon saving as can be seen in attached video are correctly processed upon restarting the editor and the assets can be used.

**Comment by LB-MateuszSzweda** [comment]
I was able to obtain Asset Processor logs from the issue affected stabilization/2110 (4d36fb9)
AP_GUI log is here:
[AP_GUI.log](https://github.com/o3de/o3de/files/7618655/AP_GUI.log)

All JobLogs can be found here:
[JobLogs.zip](https://github.com/o3de/o3de/files/7618661/JobLogs.zip)

And here is a comparison of processed groundplane512x512.fbx asset **before** updating(and freezing the Editor)
with Fbx settings tool:

[groundplane 512x512 fbx asset before updating.zip](https://github.com/o3de/o3de/files/7618686/groundplane.512x512.fbx.asset.before.updating.zip)

and **after** updating with Fbx settings tool:

[groundplane 512x512 fbx after updating.zip](https://github.com/o3de/o3de/files/7618688/groundplane.512x512.fbx.after.updating.zip)

**Comment by LB-ArturSiewniak** [comment]
I've reverified the list of affected load/save actions on the Development (366a711) build and noted the following behaviors:
#1 https://github.com/o3de/o3de/issues/5950
No longer occurs on the engine built using the official instructions.
#2 https://github.com/o3de/o3de/issues/5928
No longer occurs on the engine built using the official instructions.
#3 Creating an .inputbindings file:
Saved .inputbindings is still not visible in Editor until it is restarted. Saved .scriptcanvas files now have the same issue as well. Logged separately as https://github.com/o3de/o3de/issues/6521.
#4 Generating a physxmesh on an .fbx asset:
The Editor freeze still occurs when updating an .fbx asset with a physxmesh. Logged separately as https://github.com/o3de/o3de/issues/6522.

In short, issues #1 and #2 will no longer occur on this config/build instruction; issues #3 and #4 have both been logged separately, with a mention of the build method used. I'm closing this issue; progress on the remaining defects will be tracked in https://github.com/o3de/o3de/issues/6521 and https://github.com/o3de/o3de/issues/6522.
